Mathlib/Algebra/DirectSum/LinearMap.lean

```diff
@@ -86,4 +89,38 @@ lemma trace_comp_eq_zero_of_commute_of_trace_restrict_eq_zero
   rw [restrict_comp, trace_comp_eq_mul_of_commute_of_isNilpotent μ h_comm
     (f.isNilpotent_restrict_iSup_sub_algebraMap μ), hg, mul_zero]
 
+lemma mapsTo_biSup_of_mapsTo (s : Set ι) {f : Module.End R M} (hf : ∀ i, MapsTo f (N i) (N i)) :
+    MapsTo f ↑(⨆ i ∈ s, N i) ↑(⨆ i ∈ s, N i) := by
+  intro x hx
+  induction' hx using Submodule.iSup_induction' with i m hm y z _ _ hy hz
+  · by_cases hi : i ∈ s
+    · apply Submodule.mem_iSup_of_mem i
+      simp only [hi, iSup_pos] at hm ⊢
+      exact hf i hm
+    · replace hm : m = 0 := by simpa [hi] using hm
+      simp [hm]
+  · simp
+  · rw [map_add]
+    exact add_mem hy hz
```

Comment:
Modulo some glue to turn `MapsTo` into `map _ ≤ _`, this sounds like an easy consequence of [Submodule.map_iSup](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Submodule/Map.html#Submodule.map_iSup).

Reply:
Good idea, done!